File: src/js/directable/directables/dotmatrix/Views/ViewManager.cs
using System.Collections.Generic;

using UnityEngine;

using UnityTest.Display;
using UnityTest.UserInterface.DotMatrix.Dots;
using UnityTest.UserInterface.DotMatrix.Grid;
using UnityTest.UserInterface.DotMatrix.Shapes;

namespace UnityTest.UserInterface.DotMatrix.Views
{
    public class ViewManager
    {
        // private GridRenderer m_gridRenderer;
        private DotManager m_dotManager;
        private ShapeManager m_shapeManager;
        private List<View> m_views = new List<View>();

        // _____________________________________________________________________

        public ViewManager(GameObject container)
        {
            // Initialize Grid Data
            GridData.Initialize();

            // DEV
            // m_gridRenderer = new GridRenderer(container);

            // Create Dot Manager
            m_dotManager = new DotManager(container);

            // Create Shape Manager
            m_shapeManager = new ShapeManager(m_dotManager);

            // Dev - Create View Test
            ViewTest viewTest = new ViewTest(m_shapeManager);
            m_views.Add(viewTest);

            ViewFloor viewFloor = new ViewFloor(m_shapeManager);
            m_views.Add(viewFloor);

            ViewTextTest viewTextTest = new ViewTextTest(m_shapeManager);
            m_views.Add(viewTextTest);

            // Set Initial Resolution
            SetResolution(DisplayController.GetResolution());
        }

        // ______________________________________________________________ Update

        public void Update()
        {
            // Update All Views
            for (int i = 0; i < m_views.Count; i++)
            {
                m_views[i].Update();
            }
        }

        // __________________________________________________________ Resolution

        public void SetResolution(Vector2Int resolution)
        {
            Debug.Log("ViewManager.SetResolution: " + resolution);

            // Set Resolution
            GridData.SetResolution(resolution);

            // DEV
            // m_gridRenderer.SetResolution(resolution);

            // m_textureWrapper.SetResolution(resolution);
            m_shapeManager.SetResolution(resolution);
        }
    }
}

File: src/js/directable/dotmatrix/view/ViewManager.js
import ApplicationLogger from '../../../application/ApplicationLogger.js';

import GridData from '../grid/GridData.js';

import DotManager from '../dot/DotManager.js';
import ShapeManager from '../shape/ShapeManager.js';

import ViewTest from './ViewTest.js';
import ViewTextTest from './ViewTextTest.js';

export default class ViewManager {
	#DOT_MANAGER;
	#SHAPE_MANAGER;

	#VIEWS = [];

	#LOG_LEVEL = 4;

	// _________________________________________________________________________

	constructor(width, height) {
		ApplicationLogger.log('ViewManager', this.#LOG_LEVEL);

		// Initialize Grid Data
		GridData.initialize(width, height);

		// Create Dot Manager
		this.#DOT_MANAGER = new DotManager();

		// Create Shape Manager
		this.#SHAPE_MANAGER = new ShapeManager(this.#DOT_MANAGER);

		// Dev - Create View Test
		const VIEW_TEST = new ViewTest(this.#SHAPE_MANAGER);
		this.#VIEWS.push(VIEW_TEST);

		const VIEW_TEXT_TEST = new ViewTextTest(this.#SHAPE_MANAGER);
		this.#VIEWS.push(VIEW_TEXT_TEST);
	}

	// ____________________________________________________________________ Tick

	tick() {
		// Tick Views
		for (let i = 0; i < this.#VIEWS.length; i += 1) {
			this.#VIEWS[i].tick();
		}
	}

	// ____________________________________________________________________ Size

	setSize(width, height) {
		ApplicationLogger.log(
			'ViewManager setSize ' + width + ' ' + height,
			this.#LOG_LEVEL,
		);

		// Grid Data
		GridData.setSize(width, height);
	}
}
